### PR TITLE
REST API: Support the multipleOf JSON Schema keyword.

### PR DESCRIPTION
**Pull request by yakimun**
Added support for the multipleOf JSON Schema keyword.

Trac ticket: https://core.trac.wordpress.org/ticket/51022

### CODE REVIEW

File: src/wp-includes/rest-api.php [hunk 1]
```diff
@@ -1691,9 +1692,16 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
...
-	if ( in_array( $args['type'], array( 'integer', 'number' ), true ) && ! is_numeric( $value ) ) {
-		/* translators: 1: Parameter, 2: Type name. */
-		return new WP_Error( 'rest_invalid_param', sprintf( __( '%1$s is not of type %2$s.' ), $param, $args['type'] ) );
+	if ( in_array( $args['type'], array( 'integer', 'number' ), true ) ) {
+		if ( ! is_numeric( $value ) ) {
+			/* translators: 1: Parameter, 2: Type name. */
+			return new WP_Error( 'rest_invalid_param', sprintf( __( '%1$s is not of type %2$s.' ), $param, $args['type'] ) );
+		}
+
+		if ( isset( $args['multipleOf'] ) && ! rest_is_integer( $value / $args['multipleOf'] ) ) {
```
Comment (TimothyBJacobs) [inline src/wp-includes/rest-api.php:1701]: Can we use the modules operator instead?

File: tests/phpunit/tests/rest-api/rest-schema-validation.php [hunk 2]
```diff
@@ -435,6 +435,22 @@ public function test_null_or_integer() {
 		$this->assertSame( 'param must be between 10 (inclusive) and 20 (inclusive)', $error->get_error_message() );
 	}
 
+	/**
+	 * @ticket 51022
+	 */
+	public function test_numeric_multiple_of() {
+		$schema = array(
+			'type'       => array( 'number', 'integer' ),
+			'multipleOf' => 5,
```
Comment (TimothyBJacobs) [inline tests/phpunit/tests/rest-api/rest-schema-validation.php:444]: Let's add some tests for when `multipleOf` is a decimal.

**Comment by TimothyBJacobs** [comment]
Thanks for the patch @yakimun! I added a couple of comments.

File: src/wp-includes/rest-api.php (hunk 1 above)
Comment (yakimun) [reply src/wp-includes/rest-api.php:1701]: The modulo operator can't return decimal remainder, so I'll replace division with `fmod`.

**Comment by yakimun** [comment]
Thanks for the review @TimothyBJacobs! I have made changes according to your comments.

**Comment by TimothyBJacobs** [comment]
Committed in dbf6925. Thanks again for the patch @yakimun!

**Pull request closed without merging**